pkg/provider/okta/okta.go

```diff
@@ -1236,3 +1381,214 @@ func fidoWebAuthn(oc *Client, oktaOrgHost string, challengeContext *mfaChallenge
+	customTransport := http.DefaultTransport.(*http.Transport).Clone()
+	customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	originalTransport := oc.client.Transport
+
+	oc.client.Transport = customTransport
+
+	req, err := http.NewRequest("GET", duoCertifierURL, nil)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "error building cert validation request")
+	}
+
+	req.Header.Add("Referer", "https://"+duoHost+"/")
+	res, err := oc.client.Do(req)
+	oc.client.Transport = originalTransport
+
+	if err != nil {
+		// Local certifier not running, try online one
+		duoCertURL := fmt.Sprintf("%s?sid=%s&certs_txid=%s&type=AJAX", certUrl, url.QueryEscape(sid), txid)
+
+		req, err = http.NewRequest("GET", duoCertURL, nil)
+
+		if err != nil {
+			return nil, errors.Wrap(err, "error building cert validation request ")
+		}
+
+		req.Header.Add("Referer", "https://"+duoHost)
+		res, err = oc.client.Do(req)
+
+		if err != nil {
+			return nil, errors.Wrap(err, "error retrieving cert validation response")
+		}
+	}
+
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving body from response")
+	}
+
+	resp := string(body)
+
+	duoStat := gjson.Get(resp, "stat").String()
+	if duoStat != "OK" {
+		return nil, errors.New("error validation certificate")
+	}
+
+	certForm := url.Values{}
+	certForm.Add("sid", sid)
+	certForm.Add("certs_url", certUrl)
+	certForm.Add("certs_txid", txid)
+	certForm.Add("certifier_url", certifierUrl)
+
+	// Try POST again
+	req, err = http.NewRequest("POST", duoSubmitURL, strings.NewReader(certForm.Encode()))
+	if err != nil {
+		return nil, errors.Wrap(err, "error building authentication request")
+	}
+
+	req.URL.RawQuery = q.Encode()
+
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	res, err = oc.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving verify response")
+	}
+	defer res.Body.Close()
+
+	doc, err = goquery.NewDocumentFromReader(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "error parsing document")
+	}
+
+	return doc, nil
+}
+
+func verifyEndpointHealth(oc *Client, doc *goquery.Document, origURL string, duoEndpointHost string, duoHost string, duoSubmitURL string, postQuery url.Values) (*goquery.Document, error) {
+
+	txid, _ := doc.Find("input[name=\"txid\"]").Attr("value")
+	sid, _ := doc.Find("input[name=\"sid\"]").Attr("value")
+	ehServiceUrl, _ := doc.Find("input[name=\"eh_service_url\"]").Attr("value")
+	akey, _ := doc.Find("input[name=\"akey\"]").Attr("value")
+	responseTimeout, _ := doc.Find("input[name=\"response_timeout\"]").Attr("value")
+	parent, _ := doc.Find("input[name=\"parent\"]").Attr("value")
+	duoAppUrl, _ := doc.Find("input[name=\"duo_app_url\"]").Attr("value")
+	ehDownloadLink, _ := doc.Find("input[name=\"eh_download_link\"]").Attr("value")
+	// isSilentCollection, _ := doc.Find("input[name=\"is_silent_collection\"]").Attr("value")
```

Comment:
this wasn't present in my debug session and seems to not be necessary

Reply:
Can you remove it it? Commented out code is dead code